File: augur-engine/src/collect/rss.py
"""RSS/Atom feed collector."""

from __future__ import annotations

from datetime import datetime, timezone

import feedparser

from ..config.types import Signal


async def collect_rss(url: str) -> Signal:
    feed = feedparser.parse(url)

    items = [{
        "title": e.get("title", ""),
        "url": e.get("link", ""),
        "snippet": e.get("summary", "")[:300],
        "date": e.get("published", ""),
    } for e in feed.entries[:15]]

    return Signal(
        source="rss",
        fetched_at=datetime.now(timezone.utc).isoformat(),
        query=url,
        content={"feed_title": feed.feed.get("title", ""), "feed_url": url, "items": items},
    )

File: augur-engine/src/collect/rss.ts
import type { Signal } from '../config/types.js';
import Parser from 'rss-parser';

const parser = new Parser({
  timeout: 15_000,
  maxRedirects: 3,
});

/** Collect signals from an RSS/Atom feed */
export async function collectRss(url: string): Promise<Signal> {
  const feed = await parser.parseURL(url);

  const items = (feed.items ?? []).slice(0, 15).map((item) => ({
    title: item.title ?? '',
    url: item.link ?? '',
    snippet: item.contentSnippet?.slice(0, 300) ?? '',
    date: item.isoDate ?? item.pubDate ?? '',
  }));

  return {
    source: 'rss',
    fetchedAt: new Date().toISOString(),
    query: url,
    content: {
      feedTitle: feed.title,
      feedUrl: url,
      items,
    },
  };
}
